cmd/print.go

```diff
@@ -100,6 +101,31 @@ func printByteInfo(l int) string {
 	return fmt.Sprintf("%v %s", length.value, length.name)
 }
 
+func printSearchResult(i int, result core.Result) {
+	white := color.New(color.FgWhite).SprintFunc()
+	green := color.New(color.FgGreen).SprintFunc()
+	// NOTE: in the future we need to switch based on result.Type
+	// For now we are taking a shortcut and assuming a dataset struct
+	dsname := result.ID
+	resultStruct := &registry.Dataset{}
+	resultString := dsname
+	bytes, err := json.Marshal(result.Value)
+	if err == nil {
+		err = json.Unmarshal(bytes, resultStruct)
+		if err == nil {
+			var title, desc string
+			if resultStruct != nil {
```

Comment:
No need to check for nil here, resultStruct is both constructed a few lines above and also passed as an out parameter to Unmarshal.